src/main/java/org/folio/rest/impl/DelayedTasksAPI.java

```diff
@@ -31,4 +36,53 @@ public void deleteDelayedTaskExpiredMessages(String expirationDate, String statu
       .otherwise(this::mapExceptionToResponse)
       .onComplete(resultHandler);
   }
+
+  @Override
+  public void getDelayedTaskRetry(Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    RetryEmailsContext retryEmailsContext = new RetryEmailsContext();
+
+    Future.succeededFuture()
+      .compose(v -> lookupConfig(okapiHeaders))
+      .compose(this::checkConfiguration)
+      .map(retryEmailsContext::setConfiguration)
+      .compose(v -> findEmailsForRetry())
+      .compose(this::mapJsonObjectToEmailEntries)
+      .map(retryEmailsContext::setEmailEntries)
+      .compose(context -> CompositeFuture.all(context
+        .getEmailEntries()
+        .getEmailEntity()
+        .stream()
+        .map(email -> sendEmail(context.getConfiguration(), email)
+          .compose(this::updateEmail))
+        .collect(Collectors.toList())))
+      .map(Email.PostEmailResponse::respond200WithTextPlain)
+      .map(Response.class::cast)
```

Comment:
Do we really need this cast? `PostEmailResponse` extends `Response`.

Reply:
I guess 'yes' because we need response instead of PostEmailResponse --> Handler<AsyncResult<Response>> resultHandler